File: .tmp_py4013/batch_04_sources/2499_minimum_total_cost_to_make_arrays_unequal.js
// LeetCode 2499 - Minimum Total Cost to Make Arrays Unequal
// https://leetcode.com/problems/minimum-total-cost-to-make-arrays-unequal/

/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var minimumTotalCost = function(nums1, nums2) {
    const n = nums1.length;
    const freq = new Map();
    let ans = 0, same = 0;
    for (let i = 0; i < n; i++) {
        if (nums1[i] === nums2[i]) {
            same++;
            freq.set(nums1[i], (freq.get(nums1[i]) || 0) + 1);
            ans += i;
        }
    }
    let maxFreq = 0, maxVal = 0;
    for (const [key, value] of freq) {
        if (value > maxFreq) {
            maxFreq = value;
            maxVal = key;
        }
    }
    let need = maxFreq * 2 - same;
    if (need <= 0) return ans;
    for (let i = 0; i < n && need > 0; i++) {
        if (nums1[i] !== nums2[i] && nums1[i] !== maxVal && nums2[i] !== maxVal) {
            ans += i;
            need--;
        }
    }
    return need > 0 ? -1 : ans;
};

File: .tmp_rs4013/rust_check_batch08/2499_minimum_total_cost_to_make_arrays_unequal.rs
struct Solution;
// LeetCode 2499 - Minimum Total Cost to Make Arrays Unequal
// https://leetcode.com/problems/minimum-total-cost-to-make-arrays-unequal/

use std::collections::HashMap;

impl Solution {
    pub fn minimum_total_cost(nums1: Vec<i32>, nums2: Vec<i32>) -> i64 {
        let n = nums1.len();
        let mut freq: HashMap<i32, i32> = HashMap::new();
        let mut ans = 0i64;
        let mut same = 0;
        for i in 0..n {
            if nums1[i] == nums2[i] {
                same += 1;
                *freq.entry(nums1[i]).or_insert(0) += 1;
                ans += i as i64;
            }
        }
        let mut max_freq = 0;
        let mut max_val = 0;
        for (&v, &c) in &freq {
            if c > max_freq {
                max_freq = c;
                max_val = v;
            }
        }
        let mut need = max_freq * 2 - same;
        if need <= 0 {
            return ans;
        }
        for i in 0..n {
            if need <= 0 {
                break;
            }
            if nums1[i] != nums2[i] && nums1[i] != max_val && nums2[i] != max_val {
                ans += i as i64;
                need -= 1;
            }
        }
        if need > 0 {
            -1
        } else {
            ans
        }
    }
}

fn main() {}
